subsys/net/ip/route.h

```diff
@@ -183,12 +183,16 @@ typedef void (*net_route_cb_t)(struct net_route_entry *entry,
  */
 int net_route_foreach(net_route_cb_t cb, void *user_data);
 
+#if defined(CONFIG_NET_ROUTE_MCAST)
 /**
  * @brief Multicast route entry.
  */
 struct net_route_entry_mcast {
-	/** Network interface for the route. */
-	struct net_if *iface;
+	/** Network interfaces for the route. */
+	union {
+		struct net_if *iface;
```

Comment:
Why would we need the union, is `struct net_if *iface` used anywhere in the codebase after these changes? IMO we should avoid such ambiguities, it could be problematic in the maintenance. And it's all internal, no need to worry about backward compatibility I think.

Reply:
Hmm... I thought of backward compatibility in case if an user already has a code that adds/removes mcast routes based on events/external notifications which in case of few routes with the same prefix would require to search for a given interface with `net_route_mcast_foreach()` as `net_route_mcast_lookup()` and `net_route_mcast_del()` find the first occurence of the prefix. However, I am completely open to removing this.

Reply:
Union removed.